### PR TITLE
config: Graduate GPU featuregate to GA

### PR DESCRIPTION
**Pull request by Barakmor1**
First introduced by #2597 way back in 2019 the GPU feature gate is now deprecated with the feature state graduated to GA and as such always enabled.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
The `GPU` feature gate is now deprecated with the feature state graduated to `GA` and thus enabled by default
```

### CODE REVIEW

**Comment by xpivarc** [comment]
@Barakmor1 There are some references to removed code.

**Comment by lyarwood** [comment]
/cc

It might also be good to rebase this on https://github.com/kubevirt/kubevirt/pull/11989 that reworks FGs and keeps getting stuck at the moment.

**Comment by acardace** [comment]
@Barakmor1 please rebase this.

**Comment by Barakmor1** [comment]
@acardace Done

**Comment by Barakmor1** [comment]
> @Barakmor1 There are some references to removed code.

@xpivarc I'm not sure i understand can you elaborate please?

**Comment by fossedihelm** [comment]
@Barakmor1 I think this is not complete.
I see that `virtconfig.GPUGate` is still used in the codebase (i.e: https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/pkg/virt-api/webhooks/fuzz/fuzz_test.go#L175, https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go#L1219, https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/tests/mdev_configuration_allocation_test.go#L138), while it should not be.
Also, if this is Ga shouldn't we add entries in https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/pkg/virt-config/configuration_test.go#L774-L792?
And maybe remove the redundant `enableFeatureGate(GPUGate)` from the codebase?
Thank you

**Comment by fossedihelm** [comment]
Actually, also https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L444-L450 should be removed, and related test:
https://github.com/kubevirt/kubevirt/blob/0e1598b1d2e12eef1601e606a7be0fe4e7804a36/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go#L1204-L1216

**Comment by Barakmor1** [comment]
@fossedihelm You're right. I didn’t remove the related code. The reason is that I noticed the same thing was done with other deprecated feature gates like `Passt`, `LiveMigration`, `SRIOVLiveMigrationGate`, `NUMAEnabled`, and others. It seems we could have deleted all the code for these old feature gates, but I wasn’t sure why it hadn’t been done before. I thought the plan was to keep the feature gate logic, treat it as always enabled/disabled for now, and then remove it completely later. Maybe @EdDev or @lyarwood  can give more details about this.

**Comment by fossedihelm** [comment]
`Passt` was deprecated and not GA'ed so it is correct that nothing was changed.
Regarding the others, I cannot see any reference in the admitters, except for `NUMAEnabled` (which I think is a leftover).
Also, since we are close to the release date I would not wait for https://github.com/kubevirt/kubevirt/pull/11989, since this has more priority. Thanks

**Review by fossedihelm — APPROVED** [review]
Thank you! Nice job

File: tests/mdev_configuration_allocation_test.go [hunk 1]
```diff
@@ -214,7 +213,6 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", Serial, decorators.VGPU
 
 			By("Creating a configuration for mediated devices")
 			config = kv.Spec.Configuration
-			config.DeveloperConfiguration.FeatureGates = append(config.DeveloperConfiguration.FeatureGates, virtconfig.GPUGate)
 			config.MediatedDevicesConfiguration = &v1.MediatedDevicesConfiguration{
```
Comment (acardace) [inline tests/mdev_configuration_allocation_test.go:216]: Out of scope for this PR but I wonder if we can make this setting a default for kubevirtci and move these tests in the parallel lane..?

@xpivarc @fossedihelm

Comment (fossedihelm) [reply tests/mdev_configuration_allocation_test.go:216]: IDK if we can make such an assertion. Maybe @dhiller @brianmcarey can help here.
Consider that these tests only run in vgpu lane

Comment (Barakmor1) [reply tests/mdev_configuration_allocation_test.go:216]: I'm wondering if these tests would work on any hardware. 
Should we consider adding some decorators for these tests?

**Pull request merged**